iceoryx_utils/include/iceoryx_utils/internal/lockfree_queue/unique.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <utility>
+
+namespace iox
+{
+template <typename T>
+class unique
+{
+  public:
+    struct invalid_t
+    {
+        explicit constexpr invalid_t()
+        {
+        }
+    };
+
+    static constexpr invalid_t invalid{};
+
+    template <typename... Args>
+    unique(Args&&... args)
+        : m_value(std::forward<Args>(args)...)
+    {
+    }
+
+    unique()
+        : m_value()
+    {
+    }
+
+    unique(invalid_t)
+        : m_valid(false)
```

Comment:
This is weird behavior. Lets say I would like to create a unique with an invalid value like
```
unique<ComplexObject> bla{unique<ComplexObject>::invalid};
```
Then the default constructor of the ComplexObject is still being called and the object is constructed. This would be very inefficient and unexpected